### PR TITLE
Add script which generates a changeset for pipelinev3

### LINKED ISSUE #7504: Write `blast-radius` script which detects changed files in PR
~~This will be very similar to [Changed-Files-Functions](https://github.com/Azure/azure-rest-api-specs/blob/main/eng/scripts/ChangedFiles-Functions.ps1)~~

~~I may actually just wholesale move this file to eng/common, and add `azure-rest-api-specs` to eng/common sync.~~

This function is already present in `git-helpers.ps1`. Will re-use that. The function generating the json file at rest (with the changes) still needs to exist though. Re-purposing this issue for that.

### PR DESCRIPTION
**Pull request by scbedd**
Using `git-helpers` `Get-ChangedFiles`

Resolves #7504 

This will be used as the first step of a common build pipeline which we will add to all our repos.

image

### CODE REVIEW

File: eng/common/scripts/Generate-PR-Diff.ps1 [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+. (Join-Path $PSScriptRoot "Helpers" git-helpers.ps1)
+
+function Get-ChangedServices {
+    Param (
+        [Parameter(Mandatory=$True)]
+        [string[]] $ChangedFiles
+    )
+    
+    $changedServices = $ChangedFiles | Foreach-Object { if ($_ -match "sdk/([a-zA-Z\-]+)") { $matches[1] } } | Sort-Object -Unique
```
Comment (weshaggard) [inline eng/common/scripts/Generate-PR-Diff.ps1:30]:
```suggestion
    $changedServices = $ChangedFiles | Foreach-Object { if ($_ -match "sdk/([^/]+)") { $matches[1] } } | Sort-Object -Unique
```
Might be easier to match anything other than the next directory separator. I cannot remember for sure but does the Get-ChangedFiles always come back with forward slashes from git on both windows and linux?

**Review by weshaggard — APPROVED** [review]
Not sure what you are going to do with this info yet but looks reasonable.

File: eng/common/scripts/Generate-PR-Diff.ps1 [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#>
+[CmdletBinding()]
+Param (
+  [Parameter(Mandatory=$True)]
+  [string] $ArtifactPath,
+  [Parameter(Mandatory=$True)]
+  [string] $TargetPath
+)
+
+. (Join-Path $PSScriptRoot "Helpers" git-helpers.ps1)
```
Comment (weshaggard) [inline eng/common/scripts/Generate-PR-Diff.ps1:22]: orthogonal to this change but at some point I wanted to update these helpers to match what we have done in the specs repo at https://github.com/Azure/azure-rest-api-specs/blob/main/eng/scripts/ChangedFiles-Functions.ps1#L1. It helps us stop depending on the devops variables.

Comment (scbedd) [reply eng/common/scripts/Generate-PR-Diff.ps1:22]: Ah ok. I'll submit a PR on that + check its usages in a later PR. Thanks Wes.

File: eng/common/scripts/Generate-PR-Diff.ps1 (hunk 1 above)
Comment (scbedd) [reply eng/common/scripts/Generate-PR-Diff.ps1:30]: I will confirm this and get back to you.

Comment (scbedd) [reply eng/common/scripts/Generate-PR-Diff.ps1:30]: I mean if not it just changes to `[^/\\]`

Comment (scbedd) [reply eng/common/scripts/Generate-PR-Diff.ps1:30]: Confirmed. This is the output for a windows agent:
image

Identical.

**Pull request merged**